src/components/20-molecules/dropdown/index.js

```diff
@@ -387,11 +448,20 @@ class AXADropdown extends NoShadowDOM {
     window.addEventListener('axa-dropdown-close', this.handleWindowClick);
   }
 
-  updated() {
-    const { select, defaultTitle } = this;
+  updated(changedProperties) {
+    const { select, defaultTitle, items } = this;
     // adjust native <select>
     select.selectedIndex =
       this.findByValue(null, true) + (defaultTitle ? 1 : 0);
+
+    if (changedProperties.has('items')) {
+      // rebuild keyboard-autosuggestion data structures
+      this._autosuggestLetters = new Set();
```

Comment:
Why is this Set created outside of the `buildAutosuggestDictionary`, if that method is the only one making write operations to this set? This makes the code more unreadable, since within that method, you have to check what is actually coming, which is always an empty set. This violates the "single responsibility" principle imho.

Reply:
Fair point. Was a result of incremental development (I think). Converted `buildAutosuggestDictionary` into a proper method. Done.